include/circt/Dialect/FIRRTL/FIRRTLExpressions.td

```diff
@@ -1489,8 +1489,8 @@ def RefCastOp : FIRRTLOp<"ref.cast",
      "$input attr-dict `:` functional-type($input, $result)";
 }
 
-def RefResolveOp: FIRRTLExprOp<"ref.resolve",
-                              [RefTypeConstraint<"ref","result">]> {
+def RefResolveOp
+    : FIRRTLExprOp<"ref.resolve", [Pure, RefTypeConstraint<"ref", "result">]> {
```

Comment:
FIRRTLExprOp adds a base set of traits to those specified here.

HasCustomSSAName
InferTypeOpInterface
Pure

---> these should already be pure! (phew 👍 )

Reply:
Derp thanks!